src/Razor/test/Microsoft.VisualStudio.Editor.Razor.Test/Shared/ForegroundDispatcherWorkspaceTestBase.cs

```diff
@@ -2,11 +2,14 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using Microsoft.CodeAnalysis.Razor;
+using Microsoft.VisualStudio.Threading;
 
 namespace Xunit
 {
     public abstract class ForegroundDispatcherWorkspaceTestBase : WorkspaceTestBase
     {
         internal ForegroundDispatcher Dispatcher { get; } = new SingleThreadedForegroundDispatcher();
+
+        internal JoinableTaskContext JoinableTaskContext { get; } = new JoinableTaskContext();
```

Comment:
I forget, does VS complain about initiating joinable task contexts like this? If not then no complaints. FOr some reason I'm recalling some unique way to instantiate joinable task contexts at test time

Reply:
You were right, it wasn't working properly. I changed it so now things should be good